src/Constraint/MultiConstraint.php

```diff
@@ -117,4 +123,84 @@ public function __toString()
 
         return '[' . implode($this->conjunctive ? ' ' : ' || ', $constraints) . ']';
     }
+    /**
+     * @inheritDoc
+     */
+    public function getLowerBound()
+    {
+        $this->extractBounds();
+
+        return $this->lowerBound;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getUpperBound()
+    {
+        $this->extractBounds();
+
+        return $this->upperBound;
+    }
+
+    private function extractBounds()
+    {
+        if (null !== $this->lowerBound) {
+            return;
+        }
+
+        foreach ($this->constraints as $constraint) {
+            if (!$constraint instanceof BoundsProvidingInterface) {
+                continue; // TODO: or exception? Is it better to ignore those or not?
+            }
+
+            $constraintLower = $constraint->getLowerBound();
+            $constraintUpper = $constraint->getUpperBound();
+
+            if (null === $this->lowerBound && null === $this->upperBound) {
+                $this->lowerBound = $constraintLower;
+                $this->upperBound = $constraintUpper;
+                continue;
+            }
+
+            if ($this->versionCompare($constraintLower, $this->lowerBound, $this->isConjunctive() ? '>' : '<')) {
+                $this->lowerBound = $constraintLower;
+            }
+
+            if ($this->versionCompare($constraintUpper, $this->upperBound, $this->isConjunctive() ? '<' : '>')) {
+                $this->upperBound = $constraintUpper;
+            }
+        }
+    }
+
+    private function versionCompare(array $boundA, array $boundB, $operator)
+    {
+        return version_compare(
+            $this->prepareBoundForVersionCompare($boundA),
+            $this->prepareBoundForVersionCompare($boundB),
+            $operator
+        );
+    }
+
+    private function prepareBoundForVersionCompare(array $bound)
+    {
+        $version = str_replace(BoundsProvidingInterface::UPPER_INFINITY, (string) PHP_INT_MAX, $bound[1]);
+
+        switch ($bound[0]) {
+            case '>=':
+                $version .= '.6';
+                break;
+            case '>':
+                $version .= '.7';
+                break;
+            case '<=':
+                $version .= '.4';
+                break;
+            case '<':
+                $version .= '.3';
+                break;
```

Comment:
Can't tell if this function is smart or an atrocity :p Is this using 3/4/6/7 as a sort mechanism of some kind to make version_compare return what you want in case the two bounds are the same? I am not sure if I got it correctly or not, but this definitely needs some comments because it looks super weird.

Reply:
Pff, I did it so it must be smart 😄 I'm kidding - of course this is some black magic here and I don't really like it either. The problem is that we have to compare not only the version but also the operator. Because it does matter whether it's `>=` or `>`. That's a problem now because we went from calculating the actual bounds (+-1) in the `Constraint` to that `[operator, version]` format.
By adding another version for the operator I can just use `version_compare()` internally instead of handling it manually. But I agree that this is hacky. I mean it works but I have to find a nicer solution here. I actually do have an idea. Let me see if I can make this work.